src/model/GltfForwardTranslator.cpp

```diff
@@ -108,40 +110,49 @@ namespace model {
-  // returns : GLTF Material
-  GLTF::Material createMaterial(const std::string& name, float red, float green, float blue, float alpha, bool isDoubleSided) {
-
+  // Creates a GLTF material on the basis of raw Material Values
+  // i.e, R, G, B, A & isDoubleSided
+  // and adds to GLTF model's materials
+  // param : _materials
+  // param : mat_Data
+  // returns : index of the created GLTF Material
+  int createMaterial(std::vector<GLTF::Material>& _materials, const MaterialData& mat_Data) {
     GLTF::Material m;
-    m.name = name;
-    PbrMetallicRoughness pbr;
-    pbr.baseColorFactor = {red, green, blue, alpha};
-    pbr.metallicFactor = 0.1;
-    pbr.roughnessFactor = 0.5;
+    m.name = mat_Data.matName;
+    GLTF::PbrMetallicRoughness pbr;
+    // this contains the red, green, blue and alpha components of the main color of the material
+    pbr.baseColorFactor = {(float)(mat_Data.r / 255.0), (float)(mat_Data.g / 255.0), (float)(mat_Data.b / 255.0), (float)(mat_Data.a)};
+    // this indicates that the material should have very minimum reflection characteristics
+    // between that of a metal and non-metal material.
+    pbr.metallicFactor = 0.0;
+    // this makes the material to not be perfectly mirror-like, but instead scatter the
+    // reflected light a bit.
+    pbr.roughnessFactor = 1.0;
     m.pbrMetallicRoughness = pbr;
-    std::vector<double> eF{0, 0, 0};
-    m.emissiveFactor = eF;
-    m.alphaMode = (alpha < 1.0f) ? "BLEND" : "OPAQUE";
+    // emissive texture describes the parts of the object surface that emit light with a certain color
+    m.emissiveFactor = {0.01, 0.01, 0.00};
+    // occlusion texture can be used to simulate the effect of objects self-shadowing each other.
+    // m.occlusionTexture = "";
+    // normal map is a texture applied to modulate the surface normal in a way that makes
+    // it possible to simulate finer geometric details without the cost of a higher mesh.
+    // m.normalTexture = "";
+    m.alphaMode = (mat_Data.a < 1.0f) ? "BLEND" : "OPAQUE";
+    //alphaCutoff should not be 1.0f else MATERIAL_ALPHA_CUTOFF_INVALID_MODE	Alpha cutoff is supported only for 'MASK' alpha mode
     m.alphaCutoff = 0.5f;
-    m.doubleSided = isDoubleSided;
+    m.doubleSided = mat_Data.isDoubleSided;
 
-    return m;
+    int ret = _materials.size();
+    _materials.push_back(m);
+    return ret;
   }
 
   // Initializes the Default material for the GLTF Model materials
   // param : _materials
   void initMaterials(std::vector<GLTF::Material>& _materials) {
     // default material is always set at index 0;
-    const float grey = 0.8f;
-    const float alpha = 1.0f;
-    GLTF::Material m = createMaterial("Default material", grey, grey, grey, alpha, true);
-    _materials.push_back(m);
+    const int grey = 204;
+    const double alpha = 1.0;
+    createMaterial(_materials, MaterialData{.matName = "Default material", .r = grey, .g = grey, .b = grey, .a = alpha, .isDoubleSided = true});
```

Comment:
We're getting an error here "**use of designated initializers requires at least '/std:c++20**'"
if we're bumping up the language standard to **"ISO C++20 Standard (/std:c++20)"** from "ISO C++17 Standard (/std:c++17)" then more errors are introduced. Please Suggests...

Reply:
This worked on gcc with c++17, my bad.
Remove the designated initializers then.